internal/metrics/url_response.go

```diff
@@ -0,0 +1,57 @@
+package metrics
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	ctrlmetrics "sigs.k8s.io/controller-runtime/pkg/metrics"
+)
+
+var (
+	externalURLs = []string{"https://httpstat.us/503", "https://httpstat.us/200"}
+
+	availability = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "reference_addon_sample_availability",
+			Help: "external url availability.",
+		},
+		[]string{"url"},
+	)
+	responseTime = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "reference_addon_sample_response_time",
+			Help: "external url response time taken.",
+		},
+		[]string{"url"},
+	)
+)
+
+// RegisterMetrics must register metrics in given registry collector
+func RegisterMetrics() {
+	ctrlmetrics.Registry.MustRegister(availability)
+	ctrlmetrics.Registry.MustRegister(responseTime)
+}
+
+func RequestSampleResponseData() {
+	for _, externalURL := range externalURLs {
+		status, timeTaken := callExternalURL(externalURL)
+		availability.WithLabelValues(externalURL).Set(status)
+		responseTime.WithLabelValues(externalURL).Set(timeTaken)
```

Comment:
Usually `reponseTimes` are registered as a histogram dataType in prometheus, so you can query percentiles. A gauge is a one off status that does not really allow you to inspect the underlying distribution.

Reply:
I added sample metrics to have different values all the time so that, I can see in the time-series graph in prometheus.

Reply:
WDYT?